packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  window,
+  commands,
+  ExtensionContext,
+  workspace,
+  Uri,
+  SnippetString
+} from "vscode";
+import * as path from "path";
+import { FoamFeature } from "../types";
+import GithubSlugger from "github-slugger";
+import { TextEncoder } from "util";
+import { focusNote } from "../utils";
+
+const templatesDir = `${workspace.workspaceFolders[0].uri.fsPath}/.foam/templates`;
+
+async function getTemplates(): Promise<string[]> {
+  const templates = await workspace.findFiles(".foam/templates/**.md");
+  // parse title, not whole file!
+  return templates.map(template => path.basename(template.fsPath));
+}
+
+const feature: FoamFeature = {
+  activate: (context: ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand(
+        "foam-vscode.create-note-from-template",
+        async (context: ExtensionContext) => {
+          const templates = await getTemplates();
+          const activeFile = window.activeTextEditor?.document?.fileName;
+          const currentDir =
+            activeFile !== undefined
+              ? path.dirname(activeFile)
+              : workspace.workspaceFolders[0].uri.fsPath;
+          const selectedTemplate = await window.showQuickPick(templates);
+          const folder = await window.showInputBox({
+            prompt: `Where should the template be created?`,
+            value: currentDir
+          });
+          const title = await window.showInputBox({
+            prompt: `Enter the Title Case name for the new note`,
+            value: ``,
+            validateInput: value =>
+              value.length ? undefined : "Please enter a value!"
+          });
+          const targetFile = path.join(
+            folder,
+            `${new GithubSlugger().slug(title)}.md`
```

Comment:
what's your thinking around this? I feel we are forcing a (our) convention over users by slugifying the filename.
I wonder if it should be a setting (e.g. in my notes, the title is the filename, without slugification - same in Obsidian), or (and personally this is my preference) whether we just ask for the filename, and then it's up to the template to use the filename as it pleases (e.g. to deslugify it, or simply use it as title) - what do you think?

Reply:
The latter is the easiest to implement for now as you'll notice there's no "special" substitution support (e.g. title). I agree, probably not a good idea to force our convention.

Reply:
Should be resolved now!